data_prep.py

```diff
@@ -1,150 +1,330 @@
-import matplotlib.pyplot as plt
+import os
+from datetime import datetime
+from pathlib import Path
+
 import numpy as np
 import pandas as pd
+from loguru import logger
+
+
+class Pipeline:
+    def __init__(
+        self,
+        data_path: str = Path(os.getcwd()) / "WEC2022_Data",
+        data_type: str = "train",
+        model_type: str = "predict_upgrade",
+    ):
+        self.data_path, self.data_type, self.model_type = (
+            data_path,
+            data_type,
+            model_type,
```

Comment:
I don't think there is need for a tuple here, 
self.data_path = data_path would be more clear